src/tab/TabHeader.ts

```diff
@@ -0,0 +1,145 @@
+import { css, html, nothing, TemplateResult } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { OmniElement } from '../core/OmniElement.js';
+
+/**
+ * Control that can be used to display custom slotted content, for use within an omni-tabs component and associated omni-tab component.
+ *
+ * @import
+ * ```js
+ * import '@capitec/omni-components/tab';
+ * ```
+ * @example
+ * html```
+ * <omni-tab-header for="tab-id">
+ *  <span>Slotted Content</span>
+ * </omni-tab-header>
+ * ```
+ *
+ * @element omni-tab-header
+ *
+ * @slot - Content to render inside the component body.
+ *
+ * @cssprop --omni-tab-header-font-color - Tab header font color.
+ * @cssprop --omni-tab-header-font-family - Tab header font family.
+ * @cssprop --omni-tab-header-font-size - Tab header font size.
+ * @cssprop --omni-tab-header-font-weight - Tab header font weight.
+ *
+ * @cssprop --omni-tab-header-disabled-background-color - Tab header disabled background color.
+ * @cssprop --omni-tab-header-active-font-color - Tab header active font color.
+ *
+ * @cssprop --omni-tab-header-height - Tab header tab height.
+ * @cssprop --omni-tab-header-min-width - Tab header tab min width.
+ * @cssprop --omni-tab-header-max-width - Tab header tab max width.
+ * @cssprop --omni-tab-header-margin - Tab header tab margin.
+ *
+ * @cssprop --omni-tab-header-hover-background-color - Tab header tab hover background.
+ *
+ * @cssprop --omni-tab-header-indicator-bar-height - Tab header indicator bar height.
+ * @cssprop --omni-tab-header-indicator-bar-border-radius -  Tab header indicator bar border radius.
+ * @cssprop --omni-tab-header-indicator-bar-width - Tab header indicator bar width.
+ *
+ * @cssprop --omni-tab-header-indicator-height - Tab header indicator height.
+ * @cssprop --omni-tab-header-indicator-color - Tab header indicator color.
+ * @cssprop --omni-tab-header-indicator-border-radius - Tab header indicator border radius.
+ * @cssprop --omni-tab-header-indicator-width - Tab header indicator width.
+ */
+@customElement('omni-tab-header')
+export class TabHeader extends OmniElement {
+    /**
+     * Indicator of which omni-tab element with the matching corresponding id attribute should be displayed.
+     * @attr
+     */
+    @property({ type: String, reflect: true }) for?: string;
+
+    /**
+     * Data associated with the component.
+     * @no_attribute
+     */
+    @property({ type: Object, reflect: false }) data?: unknown;
```

Comment:
This needed?

Reply:
Yes it is used to associate a tab with a specific tab header.